core/src/main/java/net/sourceforge/jnlp/runtime/Boot.java

```diff
@@ -328,24 +321,23 @@ static URL getFileLocation() {
     }
 
     /**
-     * Gets the JNLP file from the command line arguments, or exits upon error.
+     * Fetch the JNLP file location from command line arguments. If the {@link CommandLineOptionsParser} cannot parse
+     * a {@link CommandLineOptions#JNLP} or a a {@link CommandLineOptions#HTML} it returns the first command argument.
+     *
+     * @return the file location or null if no file location can be found in the command line arguments.
      */
-    private static String getMainFile() throws InvalidArgumentException {
-        if (optionParser.getMainArgs().size() > 1
-                || (optionParser.mainArgExists() && optionParser.hasOption(CommandLineOptions.JNLP))
-                || (optionParser.mainArgExists() && optionParser.hasOption(CommandLineOptions.HTML))
-                || (optionParser.hasOption(CommandLineOptions.JNLP) && optionParser.hasOption(CommandLineOptions.HTML))) {
-            throw new InvalidArgumentException(optionParser.getMainArgs().toString());
-        } else if (optionParser.hasOption(CommandLineOptions.JNLP)) {
+    private static String getJnlpFileLocationFromCommandLineArguments() {
```

Comment:
instead of accessing the static optionParser we should add it as a param to the method - different name than 'optionParser' ;) - and work internally only with the param. By doing so we can easily test this methods with some unit tests to check that the functionality is the same as before.

Reply:
I thought about it already but wanted keep the refactoring small. But will have a look to improve to be able to write some unit tests

Reply:
Refactoring done to improve testability.